Simplex/Solver.php

```diff
@@ -17,16 +17,18 @@ class Solver
 
 	/** @var array */
 	private $steps = array();
-
-
-
-	const MAX_STEPS = 16;
+	private $max_steps = 16;
 
 
 
 	/** @param  Task $task */
-	function __construct(Task $task)
+	/** @param int $maxSteps (optional) */
+	function __construct(Task $task, int $maxSteps = null)
 	{
+		if ($maxSteps !== null) {
```

Comment:
This condition wouldn't have to be here if the default argument value was `16` instead of `null`.
Also please add re-typing to `(int) $maxSteps`.